src/models/partial_tx.ts

```diff
@@ -0,0 +1,440 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import Input from './input';
+import Output from './output';
+import Address from './address';
+import Transaction from './transaction';
+import Network from './network';
+
+import P2PKH from './p2pkh';
+import P2SH from './p2sh';
+import ScriptData from './script_data';
+
+import transaction from '../transaction';
+import helpers from '../utils/helpers';
+import { IndexOOBError, UnsupportedScriptError } from '../errors';
+
+import txApi from '../api/txApi';
+import { TOKEN_AUTHORITY_MASK, TOKEN_INDEX_MASK, HATHOR_TOKEN_CONFIG } from '../constants';
+
+type InputData = {
+  tx_id: string,
+  index: number,
+  address?: string,
```

Comment:
This is because I was trying to make a type that would work with most helpers, and only the transaction helpers require an address (and we use `transaction.signTx`).
Because of that we will always have an address so i'll make it required

Reply:
I didn't understand, to be hones. Is it going to be required? Because it's not right now.

Reply:
It will always be used on the atomic-swap, but if we later want to use this type on other features we may need to make `address` optional since some places do not use it.
I'll make it required.